src/btree/bt_ovfl.c

```diff
@@ -109,51 +121,100 @@ __ovfl_cache_col_visible(
  *	row-store: check for a globally visible update.
  */
 static bool
-__ovfl_cache_row_visible(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip)
+__ovfl_cache_row_visible(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 {
-	WT_UPDATE *upd;
-
 	/* Check to see if there's a globally visible update. */
-	for (upd = WT_ROW_UPDATE(page, rip); upd != NULL; upd = upd->next)
-		if (__wt_txn_upd_visible_all(session, upd))
+	for (; upd != NULL; upd = upd->next)
+		 if ((upd->type == WT_UPDATE_STANDARD ||
+		     upd->type == WT_UPDATE_DELETED) &&
```

Comment:
I'm nervous that this check is fragile if we add more update states, I'm also surprised that this change appears to be excluding `WT_UPDATE_RESERVED` - i.e: it's a behavior change for reserved updates, not required for overflow. Could the check be extracted out into a macro, or switched to `!WT_UPDATE_RESERVED`?